### PR TITLE
Sentry Instrumentation into Abp

### PR DESCRIPTION
**Pull request by CrazyBaran**
tl;dr;
Sometimes there is need for deep dive with Instrumentation of application to found out, what cause the problem. Automatic Sentry integration give only look into Query compiling, Database connection, Query Execution and, if we proper implement HttpClient in Abp, into HttpClient request. I would like to had full view of application also with abp dependencies.

I had hope that, @maliming didn't kill me for over excited to abp, after [Closure Allocation](https://github.com/abpframework/abp/pull/14453) but worth to try. I start develop Sentry Integration which could go deep with Instrumentation into abp framework. This is solid IT business feature, which I feel it is worthy to integrate. It could give very robust data for commercial users which look for good maintained system. My experience from asp.net zero told me that it is deserving to try. I have no longer access to my account to asp.net zero, but I remember that good integration with Relic Instrumentation, repair significant problems with couple of hard queries in our old friend asp.net boilerplate. I like open here discussion about Sentry, cause this is a stack, which I working right now. I had hope that it would be great cooperation. 
PS. I need wrote this anyway so maybe I could help other people.

Connected Issue which I found:
[https://github.com/abpframework/abp/issues/541](https://github.com/abpframework/abp/issues/541)
[https://github.com/abpframework/abp/issues/371](https://github.com/abpframework/abp/issues/371)

### CODE REVIEW

**Comment by wub** [comment]
Thanks for this @CrazyBaran. We'd definitely be interested in using something like this - currently, Sentry + ABP doesn't work very well.

**Comment by CrazyBaran** [comment]
Sorry for delay, I had still in mind to get this done, but I had storm in my work right now.

**Comment by PMExtra** [comment]
@maliming Have you reviewed this?
